.circleci/config.yml

```diff
@@ -11,6 +11,18 @@ aliases:
           sudo mkdir -p "${dirs[@]}"
           sudo chown -R circleci "${dirs[@]}"
 
+  - &restore_cache_package_json
+    # Workaround until arbitrary env var lookup is implemented by CircleCI
+    # https://discuss.circleci.com/t/cannot-use-circle-yml-environment-variables-in-cache-keys/10994/8
+    restore_cache:
+      keys:
+      - v1-npm-base.{{ checksum "_dist.txt" }}-{{ checksum "package.json" }}-{{ checksum "yarn.lock" }}
+
+  - &restore_cache_requirements
+    restore_cache:
+      keys:
+      - v1-venv-base.{{ checksum "_dist.txt" }}-{{ checksum "requirements/thumbor.txt" }}-{{ checksum "requirements/dev.txt" }}
```

Comment:
From looking https://circleci.com/docs/2.0/configuration-reference/, I think we should handle the `dist` feature via an environment variable, so that we can include the actual dist name, not its checksum, here (as per how the old code worked).

Reply:
I did that previously but found the [hard way](https://discuss.circleci.com/t/cannot-use-circle-yml-environment-variables-in-cache-keys/10994/8) that the yaml can't read from custom environment variables, except for built-in variables.

Reply:
Ugh, that's terrible.  And they don't have a way to read a file other than by hashing it, either...

Reply:
Can you send CircleCI a support ticket complaining about the fact that one can't do real variables here?  Maybe there's some undocumented solution.